types/apollo-upload-client/index.d.ts

```diff
@@ -1,22 +1,27 @@
-// Type definitions for apollo-upload-client 8.1
+// Type definitions for apollo-upload-client 14.0.0
 // Project: https://github.com/jaydenseric/apollo-upload-client#readme
 // Definitions by: Edward Sammut Alessi <https://github.com/Slessi>
+//                 Jordan Overbye <https://github.com/jordanoverbye>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.1
 
-import { ApolloLink } from "apollo-link";
-import { HttpOptions } from "apollo-link-http-common";
+import { ApolloLink, HttpOptions } from '@apollo/client';
 
-export { ReactNativeFile } from "extract-files";
+export { ReactNativeFile } from 'extract-files';
 
 declare global {
     interface GlobalFetch {
-        fetch: WindowOrWorkerGlobalScope["fetch"];
+        fetch: WindowOrWorkerGlobalScope['fetch'];
     }
 }
 
-/**
- * `createUploadLink` options match `createHttpLink` options
- * @param linkOptions `HttpOptions`
- */
-export function createUploadLink(linkOptions?: HttpOptions): ApolloLink;
+export function createUploadLink(
+    linkOptions?: HttpOptions & {
```

Comment:
thoughts on

```ts
interface UploadHttpOptions extends HttpOptions {
    // Function that checks if a value is an extractable file
    isExtractableFile?: (value: any) => boolean;
    // FormData implementation to use, defaulting to the FormData global.
    FormData?: any;
    // Customizes how extracted files are appended to the FormData instance.
    formDataAppendFile?: (
        formData: FormData,
        fieldName: string,
        file: any
    ) => void;
}
```

?

Reply:
Looks good. I've added that in and added you as a contributor